File: Java/Sivitas.java

//class Sivitas Akademik
public class Sivitas extends Mahasiswa{

	//private attribut sivitas
	private String asalUniv;
	private String email;

	Sivitas(){
		//konstruktor kosong
	}

	Sivitas(String asalUniv, String email){
		this.asalUniv = asalUniv;
		this.email = email;
	}

	// setter getter //

	public void setAsal(String asalUniv) {
		// mengeset nilai asalUniv Mahasiswa
		this.asalUniv = asalUniv;
	}

	public String getAsal() {
		//mengembalikkan nilai asalUniv mahasiswa
		return this.asalUniv;
	}

	public void setEmail(String email) {
		// mengeset nilai email Mahasiswa
		this.email = email;
	}

	public String getEmail() {
		//mengembalikkan nilai email mahasiswa
		return this.email;
	}


}

File: c++/Sivitas.cpp
#include <iostream>
#include <string>

#include "Mahasiswa.cpp"

//Using standard namespace
using namespace std;


class Sivitas : public Mahasiswa{
	//private attribut sivitas
	private:
		string asalUniv;
		string email;

	public:
		Sivitas(){
			
		}

		Sivitas(string asalUniv, string email){
			this->asalUniv = asalUniv;
			this->email = email;
		}

		// setter getter //

		void setAsal(string asalUniv) {
			// mengeset nilai asalUniv Mahasiswa
			this->asalUniv = asalUniv;
		}

		string getAsal() {
			//mengembalikkan nilai asalUniv mahasiswa
			return this->asalUniv;
		}

		void setEmail(string email) {
			// mengeset nilai email Mahasiswa
			this->email = email;
		}

		string getEmail() {
			//mengembalikkan nilai email mahasiswa
			return this->email;
		}

		~Sivitas(){
			//destruktor kosong
		}
	
};